### PR TITLE
MWPW-157100: Merch card headers as p tags for Accessibility

### PR DESCRIPTION
**Pull request by Axelcureno**
This PR changes the heading tags inside merch card to follow the following approach:
- First heading as `H3`
- All other headings as `<p>` tags.
**This is a sev1 accessibility concern.**

Screenshot 2024-10-30 at 10 12 22 AM

Resolves: [MWPW-157100](https://jira.corp.adobe.com/browse/MWPW-157100)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.live/docs/library/kitchen-sink/merch-card?martech=off
- After: https://MWPW-157100--milo--adobecom.hlx.live/docs/library/kitchen-sink/merch-card?martech=off

### CODE REVIEW

**Review by seanchoi-dev — APPROVED** [review]

**Comment by mirafedas** [comment]
@Axelcureno looks like you need to update h > p in unit tests as well

File: libs/deps/mas/merch-card.js [hunk 1]
```diff
@@ -1735,6 +1736,7 @@ merch-card [slot='heading-s'] {
     line-height: var(--consonant-merch-card-heading-s-line-height);
     margin: 0;
     color: var(--spectrum-gray-800, var(--merch-color-grey-80));
+    font-weight: 700;
```
Comment (3ch023) [inline libs/deps/mas/merch-card.js:1739]: the values below seem to repeat a lot, i wonder if its alrady worth it to put it in common place ('heading-' or smthg)
```
margin: 0;
    color: var(--spectrum-gray-800, var(--merch-color-grey-80));
    font-weight: 700;
```

**Comment by 3ch023** [comment]
Nala tests need to be fixed as well it seems

**Comment by Roycethan** [comment]
Thanks @Axelcureno  On the issues reported today
#1 is fixed - pricing below changed to 'p'
#2 per discussion will be fixable by Authors - to use correct price slot
image

**Review by npeltier — COMMENTED** [review]
in a similar direction as Mariia's: can we group common styles like
```
    margin: 0;
    color: var(--spectrum-gray-800, var(--merch-color-grey-80));
    font-weight: 700;
```

**Comment by Axelcureno** [comment]
> in a similar direction as Mariia's: can we group common styles like
> 
> ```
>     margin: 0;
>     color: var(--spectrum-gray-800, var(--merch-color-grey-80));
>     font-weight: 700;
> ```

Comment has been addressed @npeltier @3ch023.

**Review by npeltier — APPROVED** [review]

**Review by 3ch023 — APPROVED** [review]

**Comment by Roycethan** [comment]
@Axelcureno there are conflicts in files , plz check and resolve them thanks

**Comment by Axelcureno** [comment]
> @Axelcureno there are conflicts in files , plz check and resolve them thanks

conflicts resolve (one again 😅) @Roycethan

**Pull request merged**